device_flow_test.py

```diff
@@ -0,0 +1,76 @@
+import os
```

Comment:
This should be in a test class.

Reply:
Agreed, however, I think that's more of a bit of a lift for me to get sorted out on how you are handling tests. If you can take the commit of this as-is for this piece, I can commit to moving it to a test.